DiscordChatExporter.Cli.dockerfile

```diff
@@ -1,22 +1,33 @@
 # Build
-FROM mcr.microsoft.com/dotnet/sdk:7.0 AS build
+FROM mcr.microsoft.com/dotnet/sdk:7.0-alpine AS build
+
+WORKDIR /build
 
 COPY favicon.ico ./
 COPY NuGet.config ./
 COPY Directory.Build.props ./
 COPY DiscordChatExporter.Core ./DiscordChatExporter.Core
 COPY DiscordChatExporter.Cli ./DiscordChatExporter.Cli
 
-RUN dotnet publish DiscordChatExporter.Cli --configuration Release --output ./publish
+RUN dotnet publish DiscordChatExporter.Cli \
+    --self-contained \
+    --use-current-runtime \
+    --configuration Release \
+    --output ./publish
 
 # Run
-FROM mcr.microsoft.com/dotnet/runtime:7.0 AS run
+FROM mcr.microsoft.com/dotnet/runtime-deps:7.0-alpine
 
-RUN useradd dce
+RUN apk add --no-cache tzdata && \
+    mkdir -p /opt/discord_chat_exporter && \
+    adduser \
+    --disabled-password \
+    --no-create-home \
+    dce
 USER dce
+COPY --from=build /build/publish /opt/discord_chat_exporter
```

Comment:
Can we please split that `RUN` into separate commands for `apk add` and `adduser`? And ideally also add some comments above to remind future readers what they're needed for.

Also, is `mkdir` required here? I thought `COPY` creates directories.

Reply:
> Also, is `mkdir` required here? I thought `COPY` creates directories.

You are absolutely correct, it's not needed.

> Can we please split that `RUN` into separate commands for `apk add` and `adduser`?

I can do that but usually people combine as many commands into a single `RUN` instruction as possible to minimize the amount of layers, see [Dockerfile @ golang](https://github.com/docker-library/golang/blob/7b9261231ef27db073c86c4fd7339b314703fe93/1.20-rc/alpine3.17/Dockerfile) or [Dockerfile @ zookeeper](https://github.com/31z4/zookeeper-docker/blob/5cf119d9c5d61024fdba66f7be707413513a8b0d/3.8.0/Dockerfile) for example.

Reply:
> I can do that but usually people combine as many commands into a single `RUN` instruction as possible to minimize the amount of layers, see [Dockerfile @ golang](https://github.com/docker-library/golang/blob/7b9261231ef27db073c86c4fd7339b314703fe93/1.20-rc/alpine3.17/Dockerfile) or [Dockerfile @ zookeeper](https://github.com/31z4/zookeeper-docker/blob/5cf119d9c5d61024fdba66f7be707413513a8b0d/3.8.0/Dockerfile) for example.

Does it affect image size or only the build time? If it's just the latter, I think it's fine t

> I can do that but usually people combine as many commands into a single `RUN` instruction as possible to minimize the amount of layers, see [Dockerfile @ golang](https://github.com/docker-library/golang/blob/7b9261231ef27db073c86c4fd7339b314703fe93/1.20-rc/alpine3.17/Dockerfile) or [Dockerfile @ zookeeper](https://github.com/31z4/zookeeper-docker/blob/5cf119d9c5d61024fdba66f7be707413513a8b0d/3.8.0/Dockerfile) for example.

Does it affect the image size too or just the build time? If it's just the build time, I think it's fine to add a new layer since it's pretty fast as it is and I don't really build the image locally (I assume on CI the cache wouldn't matter anyway?).

Reply:
Doesn't seem to affect the image size, so I just went ahead and split them.